CMakeLists.txt

```diff
@@ -28,7 +28,8 @@ ENDIF (NOT CLANG_URL)
 set(CMAKE_MACOSX_RPATH 1)
 
 IF (NOT NUM_BUILD_THREADS)
-  SET(NUM_BUILD_THREADS 8)
+  include(ProcessorCount)
+  ProcessorCount(NUM_BUILD_THREADS)
```

Comment:
@pfultz2 Would it make sense if we separate this variable to 2:

- NUM_BUILD_THREADS sets the number of cores used for building HCC
- NUM_TEST_THREADS sets the number of cores used for testing HCC

To my empirical experience, setting NUM_BUILD_THREADS to number of available CPU cores "times 2" yields the best CPU utilization on my build cluster consisting of 3 x86 boxes.